jaimejazarenoiii/kwartracker-ios
# My Profile Page API Integration

# 📲 What
- API integration of My Profile Page

# 🤔 Why
- N/A

# 🛠 How
- Pull the latest commit/changes
- Run the application
- Signin using these credentials
_email: test_user@cody.inc_
_password: testing1234_

_To check if the data matches the database data, try query in playground_

_Note:  If you're using browser for the playground add manually the token in the cookies and the token is given for testing purposes_
1. Go to [https://staging-kwartracker.jaimejazarenoiii.me/graphiql](url)
2. Copy the code below and paste it in the first tab/box
`
mutation SignIn($signInCredential: CredentialsInput!) {
    signInWithEmail(input: {credentials: $signInCredential}) {
        token
    }
}
`
3. Copy the code below and paste it in query variables
`
{
  "signInCredential": {
    "email": "test_user@cody.inc",
    "password": "test1234"
  }
}
`
4. Copy the `token` in the output tab

5. Copy and paste the code below
     `query FetchProfile {
  profile {
    id
    firstName
    lastName
    middleName
    age
    gender
    email
  }
}`

  6. Open the console, go to **Application** tab then go to **Cookies**
  8. Add new row name it `_graphql_token` add the value which is `{"Authorization": "_token_"}` and then edit the path to `/graphiql`
  8. Do a hard refresh `command` + `shift` + `r`

  **It should look like this** 
  <img width="1440" alt="Screen Shot 2021-06-23 at 11 00 46 AM" src="https://user-images.githubusercontent.com/23580147/123028677-5970da00-d412-11eb-9d27-bf9c41831c2e.png">

  6. Run the query

  **The output should look like this**
  <img width="1437" alt="Screen Shot 2021-06-23 at 11 01 05 AM" src="https://user-images.githubusercontent.com/23580147/123028724-773e3f00-d412-11eb-9341-6fb26d945841.png">



# ✅ Acceptance criteria
- N/A

# ⏰ TODO

- [x] Rebase from develop once #41 is merge
- [x] Refactor on getting token

# 🔗 References
-
